grudge/dof_desc.py

```diff
@@ -0,0 +1,223 @@
+"""
+DOF description
```

Comment:
Maybe the `dof_desc` move should be a separate PR. That's something we could merge right away.

Reply:
Good idea, I'll create a PR for this

Reply:
@inducer I've created the PR here: https://github.com/inducer/grudge/pull/75

Note that after that gets merged, I will rebase this branch on top of that, which means I will be changing some of the commit history so the transition is cleaner (in particular, I will edit this commit: https://github.com/inducer/grudge/pull/74/commits/445e560a93ad840d55e8929f63982864d7a25b9b)